Handle warning in torchscript

albanD opened the pull request:
Fix for #25859

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27154 Handle warning in torchscript**
* #26614 add pybind version of HANDLE_TH_ERRORS
* #26613 Buffer python warning to avoid deadlocks
* #28283 Fix clang-tidy errors in csrc/Module.cpp

Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

albanD committed:
Handle warning in torchscript

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

ezyang commented on torch/csrc/jit/script/init.cpp:
```diff
@@ -702,6 +703,7 @@ void initJitScriptBindings(PyObject* module) {
               tracer::getTracingState()->graph->pop_scope();
             }
             return result;
+            END_HANDLE_TH_ERRORS_PYBIND
```
Why not the other bindings too?

albanD replied:
Because this is the only place where arbitrary pytorch code can actually run (like functions from aten). Other entry points, to the best of my knowledge only work with jit code.

albanD replied:
Because this is the only place where arbitrary pytorch code can actually run (like functions from aten). Other entry points, to the best of my knowledge only work with jit code.

ezyang replied:
But if those functions ever call `TORCH_WARN` you still need to handle unbuffering the warnings in that case, right?

albanD replied:
If they call `TORCH_WARN`, then a cpp warning will be printed (in cerr).

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD commented:
@pytorchbot retest this please

albanD commented:
@pytorchbot retest this please

albanD commented:
@pytorchbot retest this please

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

Commit:
Update on "[WIP] Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

Commit:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D17692504](https://our.internmc.facebook.com/intern/diff/D17692504)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]

albanD committed:
Update on "Handle warning in torchscript"

Fix for #25859


Differential Revision: [D18249631](https://our.internmc.facebook.com/intern/diff/D18249631)

[ghstack-poisoned]
